youtube_dl/extractor/dw.py

```diff
@@ -55,13 +68,30 @@ def _real_extract(self, url):
         title = hidden_inputs['media_title']
         media_id = hidden_inputs.get('media_id') or media_id
 
-        if hidden_inputs.get('player_type') == 'video' and hidden_inputs.get('stream_file') == '1':
-            formats = self._extract_smil_formats(
-                'http://www.dw.com/smil/v-%s' % media_id, media_id,
-                transform_source=lambda s: s.replace(
-                    'rtmp://tv-od.dw.de/flash/',
-                    'http://tv-download.dw.de/dwtv_video/flv/'))
-            self._sort_formats(formats)
+        if hidden_inputs.get('player_type') == 'video':
+            if hidden_inputs.get('stream_file') == '1':
+                formats = self._extract_smil_formats(
+                    'http://www.dw.com/smil/v-%s' % media_id, media_id,
+                    transform_source=lambda s: s.replace(
+                        'rtmp://tv-od.dw.de/flash/',
+                        'http://tv-download.dw.de/dwtv_video/flv/'))
+                self._sort_formats(formats)
+            else:
+                media = self._download_json(self._SOURCES_URL % ('v', media_id), media_id)
+                formats = []
+                for medium in media:
+                    formats.append({
+                        'url': medium.get('file'),
```

Comment:
Must not be `None`.

Reply:
> Must not be None.

If ```medium.get('file')``` returns ```None```, it would mean the underlying API has changed. From what I've understood, ```media``` cannot be ```None```. What exactly were you referring to?

Reply:
Whether it has changed or not does not mean there should be a format with invalid URL.

Reply:
Is ```ERROR: No video formats found; please report this issue...``` the preferred response? The format can be changed to be conditionally added in that case.

Reply:
Looking at the [API response](https://www.dw.com/playersources/v-44026376) using the added test case: 
```json
[ {
  "file" : "https://tvdownloaddw-a.akamaihd.net/dwtv_video/flv/vdt/2018/beng180531_004_qwibanweb_01g_sd_avc.mp4",
  "label" : "720"
}, {
  "file" : "https://tvdownloaddw-a.akamaihd.net/dwtv_video/flv/vdt/2018/beng180531_004_qwibanweb_01g_sd_sor.mp4",
  "label" : "288"
} ]
```
Every format should have ```file```. In the event of it not being present, it will fallback to the URL loaded in the page (which is still the current fallback).